odoo/addons/base/models/res_bank.py

```diff
@@ -91,9 +33,20 @@ def _get_supported_account_types(self):
     acc_holder_name = fields.Char(string='Account Holder Name', help="Account holder name, in case it is different than the name of the Account Holder", compute='_compute_account_holder_name', readonly=False, store=True)
     partner_id = fields.Many2one('res.partner', 'Account Holder', ondelete='cascade', index=True, domain=['|', ('is_company', '=', True), ('parent_id', '=', False)], required=True)
     allow_out_payment = fields.Boolean('Send Money', help='This account can be used for outgoing payments', default=False, copy=False, readonly=False)
-    bank_id = fields.Many2one('res.bank', string='Bank')
-    bank_name = fields.Char(related='bank_id.name', readonly=False)
```

Comment:
If you want to improve GUI, this should be readonly and editable only on the bank view.